controllers/storagecluster/provider_server_test.go

```diff
@@ -0,0 +1,169 @@
+		expectedDeployment := GetProviderAPIServerDeployment(instance)
+		assert.Equal(t, deployment.Spec, expectedDeployment.Spec)
+
+		service := &corev1.Service{
+			ObjectMeta: metav1.ObjectMeta{Name: ocsProviderServerName},
+		}
+		assertFound(t, r.Client, service)
+		expectedService := GetProviderAPIServerService(instance)
+		assert.Equal(t, expectedService.Spec, service.Spec)
+
+		secret := &corev1.Secret{
+			ObjectMeta: metav1.ObjectMeta{Name: ocsProviderServerName},
+		}
+		assertFound(t, r.Client, secret)
+	})
+
+	t.Run("Ensure that Deployment,Service,Secret is not created when AllowRemoteStorageConsumers is disabled", func(t *testing.T) {
+
+		r, instance := createSetupForOcsProviderTest(t, false)
+
+		obj := &ocsProviderServer{}
+		_, err := obj.ensureCreated(r, instance)
+
+		assert.NoError(t, err)
+
+		assertNotFoundProviderResources(t, r.Client)
+	})
+}
+
+func TestOcsProviderServerEnsureDeleted(t *testing.T) {
+
+	t.Run("Ensure that Deployment,Service,Secret is deleted when AllowRemoteStorageConsumers is disabled", func(t *testing.T) {
+
+		r, instance := createSetupForOcsProviderTest(t, true)
+		obj := &ocsProviderServer{}
+		// create resources and ignore error as it should be tested via TestOcsProviderServerEnsureCreated
+		_, _ = obj.ensureCreated(r, instance)
+
+		instance.Spec.AllowRemoteStorageConsumers = false
+		// the resources will be deleted through the ensureCreated func as we are not in the deletion phase
+		_, err := obj.ensureCreated(r, instance)
+		assert.NoError(t, err)
+
+		assertNotFoundProviderResources(t, r.Client)
+	})
+
+	t.Run("Ensure that Deployment,Service,Secret is deleted while uninstalling", func(t *testing.T) {
+
+		r, instance := createSetupForOcsProviderTest(t, true)
+		obj := &ocsProviderServer{}
+		// create resources and ignore error as it should be tested via TestOcsProviderServerEnsureCreated
+		_, _ = obj.ensureCreated(r, instance)
+
+		_, err := obj.ensureDeleted(r, instance)
+		assert.NoError(t, err)
+
+		assertNotFoundProviderResources(t, r.Client)
+	})
+}
+
+func assertFound(t *testing.T, cli client.Client, object client.Object) {
+
+	key := client.ObjectKeyFromObject(object)
+
+	err := cli.Get(context.TODO(), key, object)
+	assert.NoError(t, err)
+}
+
+func assertNotFound(t *testing.T, cli client.Client, object client.Object) {
+
+	key := client.ObjectKeyFromObject(object)
+
+	err := cli.Get(context.TODO(), key, object)
+	assert.Error(t, err)
+	assert.True(t, errors.IsNotFound(err))
+}
+
+func assertNotFoundProviderResources(t *testing.T, cli client.Client) {
+
+	deployment := &appsv1.Deployment{
+		ObjectMeta: metav1.ObjectMeta{Name: ocsProviderServerName},
+	}
+	assertNotFound(t, cli, deployment)
+
+	service := &corev1.Service{
+		ObjectMeta: metav1.ObjectMeta{Name: ocsProviderServerName},
+	}
+	assertNotFound(t, cli, service)
+
+	secret := &corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{Name: ocsProviderServerName},
+	}
+	assertNotFound(t, cli, secret)
+
+}
```

Comment:
This function is used by a single test and is very specific, can we just move it into the test itself? 
Right now the reader needs to jump between two distinctive locations in the source code for no real reason.